src/pages/index.tsx

```diff
@@ -2,25 +2,28 @@ import { useEffect } from 'react'
 import type { NextPage } from 'next'
 import { useRouter } from 'next/router'
 import { AppRoutes } from '@/config/routes'
-import { useHasSafes } from '@/components/welcome/MyAccounts/useAllSafes'
+import { isEmpty } from 'lodash'
+
+const ADDED_SAFES_KEY = 'SAFE_v2__addedSafes'
```

Comment:
🙅 
We have a hook called useLocalStorage that adds this prefix, please use that.

Reply:
Does the hook return undefined initially until the data is loaded from local storage? In that case we would need to find a way to stall the useEffect from running since we don't have a loading flag

Reply:
Hooks run sequentially, and localStorage itself isn't async.

Reply:
When using the useLocalStorage hook I get the same issue as there was initially, the value comes as undefined initially and it redirects to the welcome page where it should go to accounts

Reply:
I think in this case we have to access the local storage directly as otherwise the useEffect hook will run and redirect away from the index page since the initial value of what useLocalStorage returns is undefined but it will also be undefined if there is nothing in local storage so we can't differentiate that.

Reply:
Then you can use the Storage service directly, you can copy if from the the useLocalStorage hook.

Reply:
Quickly checked and it looks like we can also adjust the `useLocalStorage` hook to use an initialValue. I will open a separate refactor PR since that hook is used in a lot of places and we should run regression tests on that change.